torch/distributed/_shard/replicated_tensor.py

```diff
@@ -5,6 +5,14 @@
 from torch.distributed import distributed_c10d
 from torch.overrides import get_default_nowrap_functions
 
+_REPLICATED_WITH_NON_TENSOR_ALLOWLIST = [
+    # List of ops where if parameters are a combination of ReplicatedTensors
+    # and non-tensors, we can still return a ReplicatedTensor as the result.
+    torch.unsqueeze,
+    torch.Tensor.unsqueeze,
+    torch.Tensor.__getitem__,
```

Comment:
This is for `replicated_tensor[0]`, right?

Reply:
Yes more generally something like `replicated_tensor[...]`